test/torch_test.py

```diff
@@ -488,8 +488,39 @@ def test_send_var_with_gradient(self):
         assert model.grad.id in remote._objects
         assert model.grad.data.id in remote._objects
 
+    def test_remote_optim_step(self):
+
+        hook = TorchHook(verbose=False)
+        local = hook.local_worker
+        local.verbose = False
+        remote = VirtualWorker(id=1, hook=hook, verbose=False)
+        local.add_worker(remote)
+
+        data = Var(torch.FloatTensor([[0, 0], [0, 1], [1, 0], [1, 1]])).send(remote)
+        target = Var(torch.FloatTensor([[0], [0], [1], [1]])).send(remote)
+
+        param = []
+
+        for i in range(2):
+            torch.manual_seed(42)
+            model = nn.Linear(2, 1)
+            opt = optim.SGD(params=model.parameters(), lr=0.1)
+            model.send_(remote)
+            model.zero_grad()
+            pred = model(data)
+            loss = ((pred - target) ** 2).sum()
+            loss.backward()
+            opt.step()
+            for i in model.parameters():
+                param.append(i.get())
+            model.reset_parameters()
```

Comment:
Hey, I think the point of this test is to make sure that the parameters change after doing all this. In particular, unless I'm missing something, I think it's best to remove this line and make sure that the parameters from the second iteration _don't_ equal those from the first.

Reply:
Other than that, looks good!

Reply:
Actually i was testing that two models with same initial weights have the same updates.
but now i have changed it as you have specified.